zerver/tests/test_mattermost_importer.py

```diff
@@ -127,6 +127,35 @@ def test_process_user(self) -> None:
         self.assertEqual(user["short_name"], "snape")
         self.assertEqual(user["timezone"], "UTC")
 
+    def test_process_guest_user(self) -> None:
+        user_id_mapper = IdMapper()
+        fixture_file_name = self.fixture_file_name("guestExport.json", "mattermost_fixtures")
+        mattermost_data = mattermost_data_file_to_dict(fixture_file_name)
+        username_to_user = create_username_to_user_mapping(mattermost_data["user"])
+        reset_mirror_dummy_users(username_to_user)
+
+        sirius_dict = username_to_user["sirius"]
+        sirius_dict["is_mirror_dummy"] = False
+
+        realm_id = 3
+
+        team_name = "slytherin"
+        user = process_user(sirius_dict, realm_id, team_name, user_id_mapper)
+        self.assertEqual(user["avatar_source"], "G")
+        self.assertEqual(user["delivery_email"], "sirius@zulip.com")
+        self.assertEqual(user["email"], "sirius@zulip.com")
+        self.assertEqual(user["full_name"], "Sirius Black")
+        self.assertEqual(user["role"], UserProfile.ROLE_GUEST)
+        self.assertEqual(user["is_mirror_dummy"], False)
+        self.assertEqual(user["realm"], 3)
+        self.assertEqual(user["short_name"], "sirius")
+        self.assertEqual(user["timezone"], "UTC")
+
+        # A guest user with a `null` team value shouldn't be an admin.
```

Comment:
It says `admin` here, I assume a copy-paste leftover from the test above :smile:  Maybe this should just say e.g. `# A guest user with a `null` team value should be imported as a regular user.`? Though I can't say I understand what a user like that is supposed to mean. Perhaps @timabbott may have an idea for how to write a more helpful comment here if he recalls the background behind 69505c30a4abea2d50d563b44fd371ffd0aa9fad? Is it just that if a user doesn't have any special roles (so e.g. they're just a regular user) in any teams, then `user_dict["teams"]` may not be provided?

Reply:
I don't know why Mattermost generates those, but I think this is probably the best behavior given that reality.